i18n/create_msg.js

```diff
@@ -35,19 +40,28 @@ const CATBLOCK_MSGS = `
 Blockly.CatblocksMsgs = {};
 Blockly.CatblocksMsgs.locales = {};
 
-Blockly.CatblocksMsgs.currentLocale_ = 'en_AU';
+Blockly.CatblocksMsgs.currentLocale_ = '${DEFAULT_LANGUAGE}';
 
 Blockly.CatblocksMsgs.hasLocale = function(locale) {
   return Object.keys(Blockly.CatblocksMsgs.locales).includes(locale);
 };
 
 Blockly.CatblocksMsgs.setLocale = function(locale) {
   if (Blockly.CatblocksMsgs.hasLocale(locale)) {
-    Blockly.CatblocksMsgs.currentLocale_ = locale;
-    Blockly.Msg = Object.assign({}, Blockly.Msg, Blockly.CatblocksMsgs.locales[locale]);
+    if (Object.keys(Blockly.CatblocksMsgs.locales[locale]).length === 1) {
+      return Blockly.CatblocksMsgs.loadNewLocale(locale).then(() => {
+        Blockly.CatblocksMsgs.currentLocale_ = locale;
+        Blockly.Msg = Object.assign({}, Blockly.Msg, Blockly.CatblocksMsgs.locales[locale]);
+      });
+    } else {
+      Blockly.CatblocksMsgs.currentLocale_ = locale;
+      Blockly.Msg = Object.assign({}, Blockly.Msg, Blockly.CatblocksMsgs.locales[locale]);
+      return new Promise((resolve) => resolve());
+    }
   } else {
     // keep current locale
     console.warn('Ignoring unrecognized locale: ' + locale);
+    return new Promise((resolve, reject) => reject());
```

Comment:
use Promise().resolve() and Promise.reject() when possible

Reply:
changed from `return new Promise((resolve) => resolve());` to your `Promise().resolve()`. Also for reject.